vision/standard_object/odlc_text_detection.py

```diff
@@ -0,0 +1,149 @@
+        the bounds will all be zeros.
+    """
+
+    cropped_img: Image = crop_image(starting_image, odlc_bounds)
+
+    preprocessed_img: Image = text_detection_pre_processing(cropped_img)
+
+    detected_text: BoundingBox = text_detection(preprocessed_img)
+
+    return detected_text
+
+
+def text_detection_pre_processing(unprocessed_img: Image) -> Image:
+    """
+    This does the preprocessing effects on the passed image needed to make text detection easier.
+
+    Parameters
+    ----------
+    unprocessed_img: Image
+        The image that you want to do the preprocessing on.
+
+    Returns
+    -------
+    final_img: Image
+        The image after all of the preprocessing effects have been run.
+    """
+
+    # Contrast and brightness
+    contrast_img: Image = cv2.convertScaleAbs(unprocessed_img, alpha=1.25, beta=-30)
+
+    # Grayscale
+    grayscale_img: Image = cv2.cvtColor(contrast_img, cv2.COLOR_RGB2GRAY)
+
+    # Blur
+    blurred_img: Image = cv2.medianBlur(grayscale_img, ksize=3)
+
+    # Erode and dilate
+    kernel: Image = np.ones((5, 5), np.uint8)
+    eroded_img: Image = cv2.erode(blurred_img, kernel=kernel, iterations=1)
+    final_img: Image = cv2.dilate(eroded_img, kernel=kernel, iterations=1)
+
+    return final_img
+
+
+def text_detection(base_img: Image) -> BoundingBox:
+    """
+    Detects text on the passed image.
+
+    Parameters
+    ----------
+    base_img: Image
+        The image that you want to detect the text on.
+
+    Returns
+    -------
+    found_text: BoundingBox
+        A BoundingBox object containing the detected character as an attribute and its bounds.
+        If no text is detected, the character will be a blank string and
+        the bounds will all be zeros.
+    """
+
+    found_text: str = ""  # Text to be returned, one character
+    text_x: int = 0
+    text_y: int = 0
+    text_width: int = 0
+    text_height: int = 0
+
+    found_text_info: dict[str, list[Any]] = pytesseract.image_to_data(
+        base_img, output_type=pytesseract.Output.DICT, config="--psm 10"
+    )
+
+    text: str
+
+    for i, text in enumerate(found_text_info["text"]):
+        character: str
+        for character in text:
+            if (
+                character.isalnum() and character.isupper()
+            ):  # Text needs to be upercase and alphanumeric
+                found_text = character
+
+                # Bounding box of text in cropped image
+                text_x = found_text_info["left"][i]
+                text_y = found_text_info["top"][i]
+                text_width = found_text_info["width"][i]
+                text_height = found_text_info["height"][i]
+
+    text_bounds: Tuple[Tuple[int, int], Tuple[int, int], Tuple[int, int], Tuple[int, int]] = (
+        (text_x, text_y),
+        (text_x + text_width, text_y),
+        (text_x + text_width, text_y + text_height),
+        (text_x, text_y + text_height),
+    )
+
+    text_attributes: dict[str, Any] = {"text": found_text}
```

Comment:
Your type here for `text_attributes` should be `dict[str, str]`. Not sure if that will cause an issue with mypy because of the BoundingBox's attributes type, but if it doesn't you should change this type hint.